### PR TITLE
[feature]: show farms and pools when wallet isn't connected

### PR DESCRIPTION
**Pull request by RDMStreet**

### CODE REVIEW

File: src/modules/ModuleFarming/Pool.vue [hunk 1]
```diff
@@ -46,7 +47,9 @@ const poolSymbols = computed<TokensPair<TokenSymbol>>(() => {
 })
 
 const formattedEarned = computed(() => {
-  return formatCurrency({ amount: new BigNumber(pool.value.earned), decimals: FORMATTED_BIG_INT_DECIMALS })
+  return pool.value.earned
+    ? formatCurrency({ amount: new BigNumber(pool.value.earned), decimals: FORMATTED_BIG_INT_DECIMALS })
+    : '0'
```
Comment (0x009922) [inline src/modules/ModuleFarming/Pool.vue:52]:
```suggestion
  return formatCurrency({ amount: new BigNumber(pool.value.earned ?? 0), decimals: FORMATTED_BIG_INT_DECIMALS })
```

File: src/modules/ModuleFarming/Pool.vue [hunk 2]
```diff
@@ -114,7 +117,7 @@ usePromiseLog(withdrawState, 'farming-pool-withdraw')
 wheneverDone(withdrawState, (result) => {
   if (result.fulfilled) {
     const { earned } = result.fulfilled.value
-    const formatted = formatCurrency({ amount: earned })
+    const formatted = formatCurrency({ amount: earned! })
```
Comment (0x009922) [inline src/modules/ModuleFarming/Pool.vue:120]:
```suggestion
    invariant(earned)
    const formatted = formatCurrency({ amount: earned })
```

File: src/modules/ModuleFarming/Pool.vue [hunk 3]
```diff
@@ -196,7 +199,16 @@ function openRoiCalculator() {
...
         >
           <div
-            v-if="!enabled"
+            v-if="!dexStore.isWalletConnected"
+            class="flex items-center space-x-6"
+          >
+            <TheWalletConnect
+              v-if="!dexStore.isWalletConnected"
+              size="md"
+            />
```
Comment (0x009922) [inline src/modules/ModuleFarming/Pool.vue:208]: `<TheWalletConnect>` is intended to be used only in a single place. `The` in its name stands exactly for it.

If we need to duplicate this button, the solution to re-use this component as-is isn't ok. The logic of connection should be run in a single place without duplication.

To solve this, we should use a single component which does it, and external buttons around the app that somehow interact with this component - via Pinia or other custom storage API (custom provide/inject mechanism, for example).

File: src/modules/ModuleStaking/Pool.vue [hunk 4]
```diff
@@ -135,7 +136,7 @@ wheneverDone(withdrawState, (result) => {
   if (result.fulfilled) {
     const { earned } = result.fulfilled.value
     const formatted = formatCurrency({
-      amount: earned,
+      amount: earned!,
       symbol: { str: props.pool.rewardToken.symbol, position: 'right' },
     })
```
Comment (0x009922) [inline src/modules/ModuleStaking/Pool.vue:141]: Please use `invariant(earned)`, not `!`.

File: src/modules/ModuleStaking/Pool.vue [hunk 5]
```diff
@@ -184,7 +185,11 @@ function openRoiCalculator() {
 
     <template v-else>
       <div class="flex items-center space-x-6">
-        <template v-if="!enabled">
+        <TheWalletConnect
+          v-if="!dexStore.isWalletConnected"
+          size="md"
+        />
```
Comment (0x009922) [inline src/modules/ModuleStaking/Pool.vue:191]: Same as for Farms.

File: src/modules/ModuleStaking/composable.pools.ts [hunk 6]
```diff
@@ -0,0 +1,137 @@
...
+    })
+  })
+}
+
+function comparePools<T extends Pool>(poolA: T, poolB: T, sorting: Sorting): number {
+  switch (sorting) {
+    case Sorting.AnnualPercentageRate:
+      return poolB.annualPercentageRate.comparedTo(poolA.annualPercentageRate)
+    case Sorting.Earned:
+      return poolB.earned!.comparedTo(poolA.earned!)
```
Comment (0x009922) [inline src/modules/ModuleStaking/composable.pools.ts:116]: Please use `invariant()` for `earned`s.

**Review by 0x009922 — CHANGES_REQUESTED** [review]
LGTM, except:

- `<TheWalletConnect>` is used not as it is intended. It may cause problems. The clearer solution should be introduced.
- `earned!` in a lot of places. Non-null assertion may cause debugging problems in future until we are really sure that it will be not null. Ideally, such places should be covered with tests. Since we don't have tests, using `invariant()` runtime assertion would be the best choice, IMO.

File: src/modules/ModuleStaking/composable.pools.ts (hunk 6 above)
Comment (RDMStreet) [reply src/modules/ModuleStaking/composable.pools.ts]: Done

File: src/modules/ModuleStaking/Pool.vue (hunk 4 above)
Comment (RDMStreet) [reply src/modules/ModuleStaking/Pool.vue]: Done

File: src/modules/ModuleFarming/Pool.vue (hunk 2 above)
Comment (RDMStreet) [reply src/modules/ModuleFarming/Pool.vue]: Done

**Review by 0x009922 — APPROVED** [review]

**Pull request merged**